overwatch/src/overwatch/mod.rs

```diff
@@ -88,27 +89,66 @@ pub trait Services: Sized {
     /// # Errors
     ///
     /// The generated [`Error`].
-    fn start(&mut self, service_id: &Self::RuntimeServiceId) -> Result<(), Error>;
+    async fn start(&mut self, service_id: &Self::RuntimeServiceId) -> Result<(), Error>;
```

Comment:
Since we have a `shutdown` function which shuts down the whole overwatch, I think this method would better be renamed to `start_service`, and the same for `start_all`, `stop` and `stop_all`. But we can discuss about this. Alternatively, perhaps the `shutdown` function could be renamed to `shutdown_overwatch` or something like that.

Reply:
Oh, no. I meant it as if we should go for `start` + `shutdown_overwatch` or `start_service` + `shutdown`.
As of writing this, though, I'm clear it should be `shutdown_overwatch`.

Reply:
I like `teardown` more than `cleanup`. So I would personally go with `start_service`, `start_all_services`, `stop_service`, `teardown`, and I am not sure we would need a `stop_all_services` as well. Probably not, but maybe there might be a chance to trigger starting up again from the outside.